pkg/antctl/raw/multicluster/resourceimport.go

```diff
@@ -0,0 +1,213 @@
+	}
+	if o.namespace == "" {
+		o.namespace = metav1.NamespaceDefault
+		return
+	}
+}
+
+func init() {
+	command = &cobra.Command{
+		Use: "resourceimport",
+		Aliases: []string{
+			"resourceimports",
+			"ri",
+		},
+		Short:   "Print Multi-Cluster ResourceImports",
+		Args:    cobra.MaximumNArgs(1),
+		Example: resourceImportExample,
+		RunE:    runE,
+	}
+	o := &resourceImportOptions{}
+	options = o
+	command.Flags().StringVarP(&o.namespace, "namespace", "n", "", "Namespace of ResourceImports")
+	command.Flags().StringVarP(&o.outputFormat, "output", "o", "", "Output format. Supported formats: json|yaml")
+	command.Flags().BoolVarP(&o.allNamespaces, "all-namespaces", "A", false, "If present, list ResourceImports across all Namespaces")
+
+	GetCmd.AddCommand(command)
+}
+
+func runE(cmd *cobra.Command, args []string) error {
+	options.validateAndComplete()
+	argsNum := len(args)
+	if options.allNamespaces && argsNum > 0 {
+		return fmt.Errorf("a resource cannot be retrieved by name across all Namespaces")
+	}
+
+	kubeconfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return err
+	}
+	kubeconfig.GroupVersion = &schema.GroupVersion{Group: "", Version: ""}
+	restconfigTmpl := rest.CopyConfig(kubeconfig)
+	raw.SetupKubeconfig(restconfigTmpl)
+
+	scheme := k8sruntime.NewScheme()
+	err = mcsscheme.AddToScheme(scheme)
+	if err != nil {
+		return err
+	}
+	err = antreamcscheme.AddToScheme(scheme)
+	if err != nil {
+		return err
+	}
+	err = k8sscheme.AddToScheme(scheme)
+	if err != nil {
+		return err
+	}
+	k8sClient, err := client.New(kubeconfig, client.Options{Scheme: scheme})
+	if err != nil {
+		return err
+	}
+
+	isSingleType := false
+	if argsNum > 0 {
+		isSingleType = true
+	}
+	var res interface{}
+	switch isSingleType {
+	case true:
+		resourceImportName := args[0]
+		resourceImport := multiclusterv1alpha1.ResourceImport{}
+		err = k8sClient.Get(context.TODO(), types.NamespacedName{
+			Namespace: options.namespace,
+			Name:      resourceImportName,
+		}, &resourceImport)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return fmt.Errorf("ResourceImport %s not found in Namespace %s", resourceImportName, options.namespace)
+			}
+
+			return err
+		}
+
+		gvks, unversioned, err := k8sClient.Scheme().ObjectKinds(&resourceImport)
+		if err != nil {
+			return err
+		}
+		if !unversioned && len(gvks) == 1 {
+			resourceImport.SetGroupVersionKind(gvks[0])
+		}
+		res = resourceImport
+	case false:
+		resourceImportList := &multiclusterv1alpha1.ResourceImportList{}
+		err = k8sClient.List(context.TODO(), resourceImportList, &client.ListOptions{Namespace: options.namespace})
+		if err != nil {
+			return err
+		}
+
+		if len(resourceImportList.Items) == 0 {
+			if options.namespace != "" {
+				fmt.Printf("No resource found in Namespace %s\n", options.namespace)
+			} else {
+				fmt.Println("No resource found in all Namespaces")
+			}
+			return nil
+		}
+		res = resourceImportList.Items
+	}
+
+	switch options.outputFormat {
```

Comment:
These codes can be common for all commands? And do we have a way to share output code in antctl/command_definition.go?

Reply:
Hello Jianjun. Thanks for comment. I tried to share output code in comman_definition.go, but I found that the output method is used by commandDefinition rather than raw command, so maybe it is not easy to share the output code. And the command antctl mc get resources will list and get different resources, so maybe it's difficult to be common for all list&get commands. I don't know if you have some detailed suggestions, thank you .

Reply:
At least line 174 - 185 can be shared? For line 187 - 195, could we define an interface similar to (or even share) TableOutput in antctl/transform/common?

Reply:
okay, thanks. I'll do it.